src/transport/include/antioch/mixture_averaged_transport_evaluator.h

```diff
@@ -579,12 +579,23 @@ namespace Antioch
         }
       case(MASS_FLUX_MASS_FRACTION):
         {
-          VectorStateType molar_fractions = zero_clone(mass_fractions);
-
-          mixture.X(mixture.M(mass_fractions),mass_fractions,molar_fractions);
-
+         
+VectorStateType molar_fractions = zero_clone(mass_fractions);
```

Comment:
What happened to the indentation here?

Reply:
@klbudzin add the following to your .emacs file in your home directory:

<pre>
(setq-default indent-tabs-mode nil)
(add-hook 'before-save-hook 'delete-trailing-whitespace)
</pre>

or if you're using VIM, ask @roystgnr for the appropriate settings for .vimrc. :)

Reply:
I guess I didn't check everything all the way through. My bad, I will ad that to my .emacs file, so hopefully that doesn't happen again.

Reply:
> Secondly I converted how we calculate the diffusing mixing rule to use mole fractions instead of mass fractions. For some reason, using mass fractions gives us a different mixture molecular weight than the mole fractions. From my testing, the diffusion coefficients are very sensitive to the mixture molecular weight causing our values to be highly different from the expected values for near pure species mixture concentrations.

The mole fraction route is the way to go: I asked Sylvain to implement this but he never did(not sure why).  For further stability, Ern & G. recommend the following treatment:

For transport computations, define X_i^tr = X_i + \epsilon(number of species)^{-1} \sum_{species} X_j - X_i), where epsilon is something like machine(double) precision.  This helps with stability in the pure species limit.  It also does not change the sum of the mole fractions(up to rounding, and produces mole fractions between 0 and 1. You then compute an adjusted mass fraction Y_i^tr and transport coefficients using X_i^tr and Y_i^tr.

Reply:
I believe @SylvainPlessis did this in a PR that never got merged and is now stale (#132) :(. @klbudzin can you have a look at what is in #132 and see if it corresponds to what you have here. If not, we should get that integrated.